pkg/client/client.go

```diff
@@ -1055,11 +1055,15 @@ func (c *Client) CreateOrUpdateRoleBinding(rb *rbacv1.RoleBinding) error {
 
 	if reflect.DeepEqual(rb.RoleRef, r.RoleRef) &&
 		reflect.DeepEqual(rb.Subjects, r.Subjects) &&
-		reflect.DeepEqual(rb.Labels, r.Labels) &&
-		reflect.DeepEqual(rb.Annotations, r.Annotations) {
+		reflect.DeepEqual(rb.Labels, r.Labels) {
 		return nil
 	}
 
+	// TODO(paulfantom): currently, we don't set our own annotations for cluster role bindings.
+	// This needs a more elaborated merging logic once we do.
+	// See discussion from https://github.com/openshift/cluster-monitoring-operator/pull/932#discussion_r490089300
+	rb.Annotations = r.Annotations
```

Comment:
what happens when you set your own annoations? Having a time bomb knows to fail without unit tests or other means to tell is not a good idea.

Reply:
isn't annotation merge logic simple enough to just do it here rather than adding two TODOs plus those "means" I mention above? Seems like time well spend to make it correct once and for all.

Reply:
We are aware and I am working on a fix for this in a follow-up BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1879930. Fix would be applied to all resources, not only CRB.

Reply:
My 2 cents: if the merging logic is not too much effort, I'd pull it in as @sttts suggests. especially given the fact that 4.6 is going to be an LTS release.

@paulfantom do you estimate how much effort it would mean to pull in merging for cluster-role-bindings only? I agree that tackling all the other resources can be done separately as they don't seem to suffer the issue we're observing here.

Reply:
ok, I added merging Annotations for RBs and CRBs. Other resources will be done in follow-up PR.